# Experiment f3b4dc3d89a58a3cd2c0c4c89244db6373243780a6654bdb69155aca67775678
# Base 1c0a1345c5fe84c6fb6026cc06834dca42233eb3
diff --git a/src/taiinfmd.cpp b/src/taiinfmd.cpp
index add6fa615..cef64f7fd 100644
--- a/src/taiinfmd.cpp
+++ b/src/taiinfmd.cpp
@@ -4156,22 +4156,20 @@ void TribeInformationAIModule::removeLot(
     int typeID, unsigned char x, unsigned char y, unsigned char newStatus) {
     int idx = 0;
     if (maxBuildingLots > 0) {
-        BuildingLot* lot = buildingLots;
-        do {
+        for (; idx < maxBuildingLots; ++idx) {
+            BuildingLot* lot = &buildingLots[idx];
             if (lot->typeID == typeID && lot->x == x && lot->y == y) {
                 if (newStatus != 3) {
-                    lot->status = newStatus;
+                    buildingLots[idx].status = newStatus;
                 } else {
-                    lot->typeID = -1;
-                    lot->x = 0;
-                    lot->y = 0;
-                    lot->status = 3;
+                    buildingLots[idx].typeID = -1;
+                    buildingLots[idx].x = 0;
+                    buildingLots[idx].y = 0;
+                    buildingLots[idx].status = 3;
                 }
                 return;
             }
-            ++idx;
-            ++lot;
-        } while (idx < maxBuildingLots);
+        }
     }
 }
 
